### ISSUE TITLE
Playing a ReadableStream breaks other audioPlayers that don't use ReadableStreams

### ISSUE THREAD

**Issue by ywssp** [ywssp/Bot.wssp#63]
_Title_

Only an issue in branch [`v13-sapphire-radio`](https://github.com/ywssp/bot.wssp/tree/v13-sapphire-radio)

## Steps to Recreate bug
1. Run the `play` command inside a server
2. Once the song plays, run the `radio` command with any radio channel

## Expected Behavior
`radio` command works correctly

## Resulting behavior
`radio` command successfully connects to the websocket, but the audio will not play inside the voice channel

**Comment by ywssp** [comment]
Found out that the `audioPlayer` is causing the problem.
  
  debugging logs:
```
// ran radio command
Audio player state changed from idle to buffering.
Websocket connected
Audio player state changed from buffering to playing.

//ran leave command
Websocket closed

// ran play command

// ran radio command
Switching from YouTube to radio...
Audio player state changed from idle to buffering.
Websocket connected
(State does not change from buffering to playing)
```

**Comment by ywssp** [comment]
More testing reveals that the bug is global 

(i.e. Running the `play` command in one server results in the `radio` command not working in another server)

The `play` command will not interrupt any running radio streams, only future streams.

**Comment by ywssp** [comment]
The bug also affects radio `audioPlayer`s that are created before the `play` command is ran, that become idle.

Debugging Log
```
Created AudioPlayer for radio.
Radio Audio player state changed from idle to buffering.
Radio Audio player state changed from buffering to playing.
Created AudioPlayer for youtube.
YouTube Audio player state changed from idle to buffering.
YouTube Audio player state changed from buffering to playing.
YouTube Audio player state changed from playing to idle.
Radio Audio player state changed from playing to idle.
Radio Audio player state changed from idle to buffering.
```

**Comment by ywssp** [comment]
How does the `play` command break the `radio` command? They are setup the same.

I removed the `audioPlayer` event listeners of both commands and the `radio` command still breaks

**Comment by ywssp** [comment]
## Bug Description Update

The bug also affects other `audioPlayer`s (ex. `audioPlayer`s that play local files)

The faulty code may be from the `playInVoiceChannel` function

**Comment by ywssp** [comment]
Bug persists even if the `NoSubscriberBehavior` of the `audioPlayer` is changed, and unsubcribing the `voiceConnection`s from the `audioPlayer` does not affect it too

**Comment by ywssp** [comment]
Updated my modules and still broken
- @discordjs/voice: "^0.11.0  
- discord.js": ^13.11.0
- ytdl-core: ^4.11.2
- sodium-native: ^3.4.1

**Comment by ywssp** [comment]
# Bug Description Update
Discovered that playing ___any___ `ReadableStream` will cause every  non-`readableStream` `audioPlayer` to break

**Comment by ywssp** [comment]
# Update
It's a bug within `@discordjs/voice`, not bot.wssp

There's an [issue](https://github.com/discordjs/discord.js/issues/7232) in the Discord.js repo that describes the bug.

**Comment by ywssp** [comment]
Fixed in commit  00948ee

The command now gets a ReadableStream first from the link, and plays the received stream.

**Comment by ywssp** [comment]
Due to the the bug in `@discordjs/voice` being fixed, the workaround used in 00948ee was removed in commit 3ba23da
